lib/devices.rb

```diff
@@ -14,58 +14,79 @@ def list_devices_in_network(network_id)
+    make_api_call("/devices/#{device_serial}", :get)
   end
 
   # Update a specific device
-  # @param [String] the serial number of the device you wish to update
-  # @param [Hash] options a hash containing the attributes of the device
+  # @param [String] device_serial: the serial number of the device you wish to update
+  # @param [Hash] options: a hash containing the attributes of the device
   # additional details on accepted fields can be found in the official Meraki API Documentation (https://developer.cisco.com/meraki/api-v1/#!update-device)
   # @return [Hash] A hash with the devices full attributes after the operation has completed
   def update_device(device_serial, options)
     raise 'Options were not passed as a Hash' unless options.is_a?(Hash)
 
-    make_v1_api_call("/devices/#{device_serial}", :put, options)
+    make_api_call("/devices/#{device_serial}", :put, options)
   end
 
   # Uplink information for a specified device
   # @param [String] network_id network id where the device exists
   # @param [String] device_serial meraki serial number of the device you want to check
   # @return [Array] an array of hashes for each uplink and it's attributes
-  def get_device_uplink_stats(network_id, device_serial)
-    make_api_call("/networks/#{network_id}/devices/#{device_serial}/uplink", :get)
+  # @deprecated Please use get_organisation_device_uplinks
+  def get_device_uplink_stats(network_id, device_serial, organization_id: nil)
+    if organization_id.nil?
+      organization_id = get_network(network_id)&.dig('organizationId')
+    end
+
+    get_organization_uplink_stats(organization_id, networks: [network_id])&.detect { |device| device['serial'] == device_serial }&.dig('uplinks')
+  end
+
+  # Uplink information for a specified device
+  # @param [String] organization_id organization where the uplinks exist
+  # @param [Array] networks: an array of network ids to get uplink stats for
+  # @return [Array] an array of hashes for each device and it's uplink attributes
+  def get_organization_uplink_stats(organization_id, networks: [])
+    options = {}
+    options[:networkIds] = networks if networks.any?
+
+    make_api_call("/organizations/#{organization_id}/appliance/uplink/statuses", :get, options)
   end
 
   # Update a single devices attributes
-  # @param [String] network_id dashboard network id where the device exists
+  # @param [String] _network_id dashboard network id where the device exists (deprecated)
   # @param [String] device_serial meraki serial number of the device you want to modify
   # @param [Hash] options hash containing the attributes you want to modify.
   #   such as name, tags, longitude, latitude. A full list is found on the official Meraki API Docs
   # @return [Hash] a hash containing the devices new attribute set
-  def update_device_attributes(network_id, device_serial, options)
+  def update_device_attributes(_network_id, device_serial, options)
```

Comment:
might be worth moving it to a separate function
```ruby
# Update a single devices attributes
# @deprecated Use {#update_device_attributes} without `network_id` param instead
# @param [String] network_id dashboard network id where the device exists
# @param [String] device_serial meraki serial number of the device you want to modify
# @param [Hash] options hash containing the attributes you want to modify.
#   such as name, tags, longitude, latitude. A full list is found on the official Meraki API Docs
# @return [Hash] a hash containing the devices new attribute set
def update_device_attributes(network_id, device_serial, options)
  ...
end

# Update a single devices attributes
# @param [String] device_serial meraki serial number of the device you want to modify
# @param [Hash] options hash containing the attributes you want to modify.
#   such as name, tags, longitude, latitude. A full list is found on the official Meraki API Docs
# @return [Hash] a hash containing the devices new attribute set
def update_device_attributes(device_serial, options)
  ...
end
```

Reply:
how's that?